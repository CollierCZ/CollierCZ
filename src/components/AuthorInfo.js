import { Link } from "gatsby";
import { shape, string } from "prop-types";
import React from "react";

const AuthorInfo = ({author}) => {
  const url = author.url.value;
  const name = author.name.value;
  const bio = author.short_bio.value;
  return (
    <>
      <h4>
        <Link to={"/"+url}>{name}</Link>
      </h4>
      <div className="author-bio" dangerouslySetInnerHTML={{ __html: bio}}>
      </div>
    </>
  );
}

AuthorInfo.propTypes = {
  author: shape({
    name:shape({
      value: string.isRequired
    }).isRequired,
    short_bio:shape({
      value: string.isRequired
    }).isRequired,
    url: shape({
      value: string.isRequired
    }).isRequired
  })
}

export default AuthorInfo;
