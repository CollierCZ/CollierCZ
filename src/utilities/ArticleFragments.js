import { graphql } from "gatsby"

export const ArticleListFragment = graphql`
  fragment ArticleListFragment on KontentItemArticle {
    fields {
      slug
      tags
      date
    }
    elements {
      metadata__description {
        value
      }
      metadata__keywords {
        value
      }
      categories {
        value {
          name
        }
      }
      title {
        value
      }
      teaser {
        value {
          url
        }
      }            
    }
  }
`