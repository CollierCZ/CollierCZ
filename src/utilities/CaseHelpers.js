const slugify = (text) => {
  return text
    .toString()
    .toLowerCase()
    .trim()
    .replace(/\s+/g, '-')       // Replace spaces with -
    .replace(/[^\w-]+/g, '')     // Remove all non-word chars
}

const titleCase = (text) => {
  return(
    text.replace(/\w\S*/g, function(txt) {
      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    })
  )
}

module.exports =
{
  slugify: slugify,
  titleCase: titleCase
}