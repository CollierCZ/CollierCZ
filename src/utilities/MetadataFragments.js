import { graphql } from "gatsby"

export const MetadataFragment = graphql `
  fragment MetadataFragment on KontentItemHome {
    elements {
      title {
        value
      }
      splash_image {
        value {
          url
        }
      }
      blog_logo {
        value {
          url
        }
      }
      metadata__description {
        value
      }
      socialmedia {
        value
      }
      base_url {
        value
      }
    }
  }
`

export const SocialMediaFragment = graphql `
  fragment SocialMediaFragment on KontentItemHome {
    elements {
      socialmedia {
        value
      }
    }
  }
`