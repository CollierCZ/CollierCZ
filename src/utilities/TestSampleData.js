export const author = {
  short_bio: {
    value:
      '<p>Originally from <a href="https://en.wikipedia.org/wiki/Lexington,_Massachusetts" data-new-window="true" target="_blank" rel="noopener noreferrer">Lexington</a> and now residing in <a href="https://www.gotobrno.cz/en/" data-new-window="true" target="_blank" rel="noopener noreferrer">Brno</a>, Aaron is an educator, editor, and elucidator.</p>'
  },
  picture: {
    value: [
      {
        url:
          "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/f7377b84-8ff0-4b77-bc0f-cfdc34d379cc/quixote.jpg"
      }
    ]
  },
  name: {
    value: "Aaron Collier"
  },
  url: {
    value: "aaron-collier"
  }
};

export const simpleImage = {
  description: "A cool picture",
  url: "https://example.com/"
};

export const sampleQuote = {
  system: {
    type: "quote"
  },
  elements: {
    quote: {
      resolvedData: {
        html:
          "<p>You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.</p>"
      }
    },
    source: {
      resolvedData: {
        html:
          "<p>Atticus Finch in <em>To Kill a Mockingbird </em>by Harper Lee</p>"
      }
    }
  }
};

export const sampleShowcase = {
  system: {
    type: "showcase"
  },
  elements: {
    items: {
      linked_items: [
        {
          id: "457d3938-5907-5cca-947d-bae1583e105d",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=vj7TyUgBEW8"
            },
            name: {
              value: "Write the Docs talk"
            },
            short_description: {
              links: [
                {
                  linkId: "3cc9d486-53e2-495d-9e2b-25665e0ff29f",
                  urlSlug: "seeing-your-docs-through-different-eyes",
                  type: "article",
                  codename: "write_the_docs_2019"
                }
              ],
              resolvedData: {
                html:
                  '<p>A video of a talk I gave at the <a href="http://www.writethedocs.org/conf/prague/2019/">Write the Docs Prague 2019</a> conference. Or read a <a data-item-id="3cc9d486-53e2-495d-9e2b-25665e0ff29f" href="">text version of the talk</a>.</p>'
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/51e9be55-8502-40ca-b374-a1305b36dc56/Slide16.PNG"
                }
              ]
            }
          },
          fields: {
            slug: "write-the-docs-talk"
          }
        },
        {
          id: "fce4fe5b-59c9-5c88-8634-16b63df49d4a",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=MlMkAqxoPZM"
            },
            name: {
              value: "Structured content video"
            },
            short_description: {
              links: [],
              resolvedData: {
                html:
                  "<p>A video about what structured content is and how it can benefit content creators.</p>"
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/b0481084-18a8-4cf0-ae01-d1a0dbd766ce/structured_content.jpg"
                }
              ]
            }
          },
          fields: {
            slug: "structured-content-video"
          }
        }
      ]
    }
  }
};
