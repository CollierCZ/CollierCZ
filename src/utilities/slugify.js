let slugify = (text) => {
    return text
        .toString()
        .toLowerCase()
        .trim()
        .replace(/\s+/g, '-')           // Replace spaces with -
        .replace(/[^\w-]+/g, '')       // Remove all non-word chars
}


module.exports.slugify = slugify