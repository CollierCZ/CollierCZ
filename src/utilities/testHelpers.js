import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { navigate } from "gatsby";

export const simpleRenderer = component => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
};
export const clickTextLink = component => {
  const mountedComponent = mount(component);
  mountedComponent
    .find("a")
    .first()
    .simulate("click");
  expect(navigate).toHaveBeenCalled();
};
