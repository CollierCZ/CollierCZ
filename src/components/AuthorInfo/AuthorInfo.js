import React from "react";
import { Link } from "gatsby";
import "./AuthorInfo.css";
import AuthorMeta from "../AuthorMeta/AuthorMeta";
import AuthorLink from "../AuthorLink/AuthorLink";

const Bio = props => {
  const { bio, moreArticlesUrl } = props;
  if (bio) {
    return <div dangerouslySetInnerHTML={{ __html: bio}} />;
  }
  return (
    <p>
      Read <Link to={moreArticlesUrl}>more articles</Link> by this author.
    </p>
  );
};

class AuthorInfo extends React.Component {
  render() {
    const { prefix } = this.props;
    const id = this.props.author.url.value;
    const name = this.props.author.name.value;
    const image = this.props.author.picture.value[0].url;
    const bio = this.props.author.short_bio.value;
    const authorInfoUrl = prefix ? `${prefix}/${id}` : id;
    if (image) {
      return (
        <section className="author">
          <h4>
            <AuthorLink url={authorInfoUrl} name={name} />
          </h4>
          <Bio bio={bio} moreArticlesUrl={authorInfoUrl} />
          <AuthorMeta />
        </section>
      );
    }
    return null;
  }
}

export default AuthorInfo;
