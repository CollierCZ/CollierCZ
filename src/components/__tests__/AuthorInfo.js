import React from "react"
import AuthorInfo from "../AuthorInfo"
import { simpleShallowRender } from "../../utilities/testHelpers"
import { author } from "../../utilities/TestSampleData"

describe("AuthorInfo", () => {
  it("renders correctly", () => {
    simpleShallowRender(<AuthorInfo author={author} />)
  })
})