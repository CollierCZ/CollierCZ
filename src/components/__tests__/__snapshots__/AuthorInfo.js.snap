// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorInfo renders correctly 1`] = `
Array [
  .c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c2,
.c2:link,
.c2:visited {
  color: #252A31;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c2:hover,
.c2:active,
.c2:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c3 a:not(.c1),
.c3 a:not(.c1):link,
.c3 a:not(.c1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c3 a:not(.c1):hover,
.c3 a:not(.c1):active,
.c3 a:not(.c1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 500;
  color: #252A31;
  line-height: 20px;
  margin: 0;
  margin-bottom: 8px;
}

<div
    className="c0"
  >
    <a
      className="c1 c2"
      href="/aaron-collier"
      onClick={[Function]}
    >
      Aaron Collier
    </a>
  </div>,
  .c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c2,
.c2:link,
.c2:visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c2:hover,
.c2:active,
.c2:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
  margin-bottom: 8px;
}

.c0 a:not(.c1),
.c0 a:not(.c1):link,
.c0 a:not(.c1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c0 a:not(.c1):hover,
.c0 a:not(.c1):active,
.c0 a:not(.c1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<p
    className="c0"
  >
    Originally from 
    <a
      className="c1 c2"
      href="https://en.wikipedia.org/wiki/Lexington,_Massachusetts"
      onClick={[Function]}
    >
      Lexington
    </a>
     and now residing in 
    <a
      className="c1 c2"
      href="https://www.gotobrno.cz/en/"
      onClick={[Function]}
    >
      Brno
    </a>
    , Aaron is an educator, editor, and elucidator.
  </p>,
]
`;
